legacy-fabric-logger-api-v1/common/src/main/java/net/legacyfabric/fabric/impl/logger/LoggerHelper.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2020 - 2022 Legacy Fabric
+ * Copyright (c) 2016 - 2022 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.legacyfabric.fabric.impl.logger;
+
+import net.legacyfabric.fabric.api.logger.v1.Logger;
+
+public class LoggerHelper {
+	public static final String API = "LegacyFabricAPI";
+
+	public static Logger getLogger(String context, String... subs) {
+		try {
+			return new NativeLoggerImpl(context, subs);
+		} catch (NoClassDefFoundError e) {
```

Comment:
this is cursed. we should probably do this check only once rather than creating an error each time.

e.g.
```java
private static TriState exists = TriState.DEFAULT;
private static BiFunction<...> javaLogger = ...;
private static BiFunction<...> nativeLogger = ...;

// ...

if (exists == TriState.DEFAULT) {
    try {
        Class.forName("theClassName");
        exists = TriState.TRUE;
    } catch (ClassNotFoundException e) {
        exists = TriState.FALSE;
    }
}

return exists.get() ? nativeLogger.apply(...) : javaLogger.apply(...);
```

Reply:
Good catch 👍, I'll do that